OpenEXR/IlmImf/ImfDwaCompressor.cpp

```diff
@@ -1937,7 +1937,7 @@ DwaCompressor::compress
         _outBufferSize = outBufferSize;
         if (_outBuffer != 0)
             delete[] _outBuffer;       
-        _outBuffer = new char[outBufferSize];
+        _outBuffer = new char[outBufferSize]();
```

Comment:
Holy moly, I have been using C++ for a long time and this is not a trick I knew!

Reply:
Very interesting! I had to investigate the distinction...

new char[] performs default initialization:
Per https://en.cppreference.com/w/cpp/language/default_initialization
"if T is an array type, every element of the array is default-initialized;"
but the elements are POD types, for which default initialization means "nothing is done".

new char[]() performs value initialization (since C++03):
Per https://en.cppreference.com/w/cpp/language/value_initialization
"The effects of value initialization are: [...]
3) if T is an array type, each element of the array is value-initialized;"
then for POD types, value-initialization with empty arguments results in zero-initialization (since C++11):
https://en.cppreference.com/w/cpp/language/zero_initialization
"The effects of zero initialization are:
If T is a scalar type, the object's initial value is the integral constant zero explicitly converted to T."

Questions for @peterhillman @kdt3rd @larrygritz et al:
1) Is this method performance-critical? Since this fix will now initialize all bytes, many of which will subsequently be overwritten. (There may not be a way anyhow to determine the bytes skipped during compression that still need initialization.)
2) In the case when _outbuffer is allocated, doesn't this change obviate the subsequent call to memset to zero the chunk header?
line 1983 "memset (_outBuffer, 0, NUM_SIZES_SINGLE * sizeof (Int64));"